syntax = "proto3";

import "EchoAttributes.proto";

package message_communication_security;

message symmetricKeyMaterial
{
    bytes key = 1 [(bytes_size) = 16];
    bytes iv = 2 [(bytes_size) = 16];
}

service SymmetricKeyEstablishment
{
    option (service_id) = 1;

    rpc ActivateNewKeyMaterial(symmetricKeyMaterial) returns (Nothing) { option (method_id) = 1; }
}
